### PR TITLE
feat(gatsby-plugin-sharp): Add tracedSVG option in fluid and fixed processors

### PR DESCRIPTION
**Pull request by oorestisime**
Fix: #11912

### CODE REVIEW

**Comment by wardpeet** [comment]
Mind adding some tests for it? It's good enough to just test if traceSVG is ran with the correct values. And not ran if it's not needed ^^
https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/src/__tests__/index.js

**Comment by oorestisime** [comment]
Oups sorry didn't pay attention to tests.

I also kind of avoided duplicate code from my first commit. i am not sure though if it is  explicit enough. or maybe a better function name. let me know

File: packages/gatsby-plugin-sharp/src/__tests__/index.js [hunk 1]
```diff
@@ -319,6 +319,53 @@ describe(`gatsby-plugin-sharp`, () => {
...
+        width: 100,
+        tracedSVG: { color: `#FF0000` },
+      }
+
+      let result = await fixed({
+        file,
+        args,
+      })
+
+      expect(result.tracedSVG).toBeUndefined()
```
Comment (sidharthachatterjee) [inline packages/gatsby-plugin-sharp/src/__tests__/index.js]: I'd add a jest mock for the `traceSVG` function (it's already exported) and add an assertion that it's not called. What do you think?

Comment (oorestisime) [reply packages/gatsby-plugin-sharp/src/__tests__/index.js]: Any way i could mock this properly?

thought about

```
const spy = jest.spyOn(rest, `traceSVG`)

    beforeEach(() => {
      spy.mockClear()
    })

    afterAll(() => {
      spy.mockClear()
    })
```

with rest

```
const {
  base64,
  fluid,
  fixed,
  queueImageResizing,
  getImageSize,
  ...rest
} = require(`../`)
```

Comment (sidharthachatterjee) [reply packages/gatsby-plugin-sharp/src/__tests__/index.js]: This seems trickier than I initially thought

Guess we should merge and iterate on the tests and the code (to make it easier to test) later

**Comment by sidharthachatterjee** [comment]
Published in gatsby-plugin-sharp@2.0.22

**Pull request merged**